plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/route/RouteChooser.java

```diff
@@ -41,11 +41,7 @@ public RouteChooser(Shell shell) {
 	public IRoute chooseRoute(List<IRoute> routes) {
 		final IRoute[] selectedRoute = new IRoute[1];
 		Display.getDefault().syncExec(() -> {
-			if (shell == null) {
```

Comment:
what's the point in having a `shell` field in this class ? Why not just call this `PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell()` when a shell is needed, instead ?